### PR TITLE
[14.0] [IMP] purchase_allowed_product: global setting and access rights

### PR DESCRIPTION
**Pull request by renda-dev**
- Add a global setting in purchase: "**Use only allowed products** by default"
    - When creating a new PO, field "**Use only allowed products**" is enabled by default
- Add a configuration in user: "Hide "**Use only allowed products**" in PO and invoice"
    - If this field is enabled in user, field "**Use only allowed products**" is not visible for this user.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Your can do that setting a default, and the configuration settings have a direct option for that. Check other examples in Odoo core.

**Comment by renda-dev** [comment]
Hi @pedrobaeza, that's awesome, thank you so much for letting me know :)

**Comment by francesco-ooops** [comment]
@renda-dev there is something strange in settings as I don't see the setting and the "Purchase" section is duplicated

image

**Review by francesco-ooops — APPROVED** [review]
Functional ok!

the error above is due to some other module (will try to find the culprit)

**Review by geomer198 — APPROVED** [review]
LGTM

**Comment by francesco-ooops** [comment]
@pedrobaeza good for merge?

File: purchase_allowed_product/models/supplied_product_mixin.py [hunk 1]
```diff
@@ -8,15 +8,27 @@ class SuppliedProductMixin(models.AbstractModel):
...
+        'Hide "Use only allowed products" in PO and invoice',
+        default=_default_hide_use_only_supplied_product,
+        readonly=True,
+    )
+
     @api.onchange("partner_id")
     def _onchange_partner_id_supplied_product(self):
         self.use_only_supplied_product = (
-            self.partner_id.use_only_supplied_product
+            self.use_only_supplied_product
```
Comment (pedrobaeza) [inline purchase_allowed_product/models/supplied_product_mixin.py]: With this, you are avoiding the option to disable the check automatically if previously was set. This is an undesired behavior.

File: purchase_allowed_product/models/supplied_product_mixin.py [hunk 2]
```diff
@@ -8,15 +8,27 @@ class SuppliedProductMixin(models.AbstractModel):
     _name = "supplied.product.mixin"
     _description = "Mixin for documents that want to restrict products"
 
+    def _default_hide_use_only_supplied_product(self):
```
Comment (pedrobaeza) [inline purchase_allowed_product/models/supplied_product_mixin.py]: Instead of doing this, just put the corresponding `groups` in the view.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by francesco-ooops — APPROVED** [review]
Functionally still good, @pedrobaeza what do you say?

**Comment by francesco-ooops** [comment]
@renda-dev rebase please

**Comment by francesco-ooops** [comment]
@pedrobaeza shall we go? :)

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge minor

**Pull request merged**